packages/bibliography/src/main.rs

```diff
@@ -0,0 +1,358 @@
+}
+
+macro_rules! push_citation {
+    ($e:expr) => {module!("list-push", $e, {"name": "inline_citations"})}
+}
+
+macro_rules! add_citation_label {
+    ($a:expr, $b:expr) => {module!("set-add", format!("{}{}", $a, $b), {"name": "inline_citation_labels"})}
+}
+
+macro_rules! cite_label {
+    ($e:expr) => {
+        module!("cite-internal-do-not-use", $e)
+    };
+}
+
+macro_rules! text {
+    ($e:expr) => {
+        module!("__text", $e)
+    };
+}
+
+fn manifest() {
+    print!(
+        "{}",
+        json!(
+            {
+            "name": "bibliography",
+            "version": "0.1",
+            "description": "This package supports bibliographies and in-text citations.",
+            "transforms": [
+                {
+                    "from": "cite",
+                    "to": ["any"],
+                    "arguments": [
+                        {"name": "note", "default": "", "description": "Note to show after citation, such as p. 3"}
+                    ],
+                    "description": "Add an inline citation to one of the sources defined in the bibliography. Example: [cite] modmark",
+                    // "variables": {
+                        // We really want this, but we can't since it would give cyclic dependencies
+                        // together with unknown-content. So, we disable unknown-content
+                        // "inline_citations": {"type": "list", "access": "push"}
+                    // },
+                    "unknown-content": true // This gives cyclic dependencies if enabled together
+                    // with inline_citations/list-push, see below
+                },
+                {
+                    "from": "cite-internal-do-not-use",
+                    "to": ["any"],
+                    "arguments": [],
+                    "description": "Do not use this module",
+                    "variables": {
+                        "inline_citation_labels": {"type": "set", "access": "read"}
+                    }
+                },
+                {
+                    "from": "bibliography",
+                    "to": ["any"],
+                    "description": "Inserts a bibliography into the document. \
+                    This module needs to exist for [cite]s to work properly. \
+                    The bibliography may be read from the BibLaTeX or Hayagriva YAML format, \
+                    from either the body of this module or a file passed in as the \"file\" argument.",
+                    "arguments": [
+                        {"name": "style", "default": "IEEE", "type": ["IEEE", "APA", "MLA", "CMoS"], "description": "The style to have the bibliography in"},
+                        {"name": "file", "default": "", "description": "A file containing BibLaTeX or Hayagriva YAML with the bibliography"},
+                        {"name": "visibility", "default": "shown", "type": ["shown", "hidden"], "description": "Whether the bibliography is 'shown' or 'hidden'. \
```

Comment:
Maybe worth using the same names as the CSS property `visibility` does?
```suggestion
                        {"name": "visibility", "default": "shown", "type": ["visible", "hidden"], "description": "Whether the bibliography is 'shown' or 'hidden'. \
```